src/Idris/Pretty.idr

```diff
@@ -276,10 +278,22 @@ mutual
       go d (PDotted _ p) = dot <+> go d p
       go d (PImplicit _) = "_"
       go d (PInfer _) = "?"
-      go d (POp _ op x y) = parenthesise (d > appPrec) $ group $ go startPrec x <++> pretty op <++> go startPrec y
+      go d (POp _ op x y) = parenthesise (d > appPrec) $ group $ go startPrec x <++> prettyOp <++> go startPrec y
+        where prettyOp : Doc ann
+              prettyOp = if isOpName op
+                then pretty op
+                else Chara '`' <+> pretty op <+> Chara '`'
       go d (PPrefixOp _ op x) = parenthesise (d > appPrec) $ pretty op <+> go startPrec x
-      go d (PSectionL _ op x) = parens (pretty op <++> go startPrec x)
-      go d (PSectionR _ x op) = parens (go startPrec x <++> pretty op)
+      go d (PSectionL _ op x) = parens (prettyOp <++> go startPrec x)
+        where prettyOp : Doc ann
+              prettyOp = if isOpName op
+                then pretty op
+                else Chara '`' <+> pretty op <+> Chara '`'
+      go d (PSectionR _ x op) = parens (go startPrec x <++> prettyOp)
+        where prettyOp : Doc ann
```

Comment:
Can we lift `prettyOp` as a definition on the same level as `go` so
that we don't need to have 3 copies of the code?

Reply:
thx, fixed